libs/generic-view/store/src/lib/selectors/data-migration-devices.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { ReduxRootState } from "Core/__deprecated__/renderer/store"
+import { createSelector } from "@reduxjs/toolkit"
+import pureBlackImage from "Core/__deprecated__/renderer/images/pure-black-render.png"
+import pureGreyImage from "Core/__deprecated__/renderer/images/pure-gray-render.png"
+import { kompaktImg } from "Root/demo-data/kompakt-img"
+import { intl } from "Core/__deprecated__/renderer/utils/intl"
+import { defineMessages } from "react-intl"
+
+const messages = defineMessages({
+  "0e8d.2006": {
+    id: "devices.0e8d.2006.name",
+  },
+})
+
+export const selectDataMigrationSourceDevices = createSelector(
+  (state: ReduxRootState) => state.deviceManager.devices,
+  (devices) =>
+    devices
+      .filter(({ deviceType }) => deviceType === "MuditaPure")
+      .map((device) => ({
+        name: device.deviceType!,
+        image: device.caseColour === "black" ? pureBlackImage : pureGreyImage,
+        serialNumber: device.serialNumber!,
+      }))
+)
+
+export const selectDataMigrationTargetDevices = createSelector(
+  (state: ReduxRootState) => state.genericViews.devicesConfiguration,
+  (devices) => {
+    return Object.values(devices)
+      .map((device) => {
+        if (!device.apiConfig) return undefined
+        const name = intl.formatMessage(
```

Comment:
I'm not sure if translations should be resolved in selectors. Components or custom hooks fit the use case better.

Reply:
Fixed in [8bc91e0](https://github.com/mudita/mudita-center/pull/1885/commits/8bc91e081b6627edb65e7d1eed2e7b9b0862cc1f)